master/buildbot/steps/shellsequence.py

```diff
@@ -63,10 +65,11 @@ def validateAttributes(self):
 
     @defer.inlineCallbacks
     def getRenderingFor(self, build):
+        rv = copy.copy(self)
         for p_attr in self.publicAttributes:
             res = yield build.render(getattr(self, p_attr))
-            setattr(self, p_attr, res)
-        defer.returnValue(self)
+            setattr(rv, p_attr, res)
+        defer.returnValue(rv)
```

Comment:
Fix looks good. We require the writing of unit tests for bugs fixed.

http://docs.buildbot.net/latest/developer/tests.html